### ISSUE TITLE
type error when using 'render:html' hook in nitro plugin

### ISSUE THREAD

**Issue by mkarras** [nuxt/nuxt#21693]
### Environment

- Operating System: Windows_NT
- Node Version:     v18.12.1
- Nuxt Version:     3.5.3
- Nitro Version:    2.5.0
- Package Manager:  npm@8.19.2
- Builder:          vite
- User Config:      ssr, telemetry, runtimeConfig, app, typescript, build, alias, vite, nitro, hooks, modules, routeRules, css, devtools, devServer
- Runtime Modules:  @nuxtjs/tailwindcss@6.8.0, @nuxtjs/i18n@8.0.0-beta.10, @nuxt/devtools@0.4.6

### Reproduction

Use the `render:html` hook from Nuxt inside a nitro plugin hook (like described in https://nuxt.com/docs/guide/going-further/hooks#usage-within-a-nitro-plugin) by creating a file with:
`server/plugins/xyz.ts`
```
export default defineNitroPlugin((nitroApp) => {
    nitroApp.hooks.hook('render:html', (html: NuxtRenderHTMLContext, { event }: { event: H3Event }) => {
                         ^^^^^^^^^^^
     ...
});
```

### Describe the bug

After updating from Nuxt `3.5.1` to `3.5.3` TS is throwing:
```
error TS2345: Argument of type '"render:html"' is not assignable to parameter of type 'HookKeys<NitroRuntimeHooks>'.
```
This is caused by the newly introduced typing with https://github.com/unjs/nitro/pull/1316/files#r1233083925

Nuxt is using custom hooks (like `render:html`) but there are not part of the typing (it is not extensible).

### Additional context

https://github.com/nuxt/nuxt/blob/71ef8bd3ff207fd51c2ca18d5a8c7140476780c7/packages/nuxt/src/core/runtime/nitro/renderer.ts#L228

**Comment by danielroe** [comment]
This is pending a Nitro patch release 👍

**Comment by danielroe** [comment]
This should be resolved in the [edge channel](https://nuxt.com/docs/guide/going-further/edge-channel/), or in the next release.

Let me know if not and I'll reopen.

**Comment by mkarras** [comment]
Fixed with Nuxt `3.6`.
Thank you @danielroe !
